app/routes/submissions.rb

```diff
@@ -15,8 +15,15 @@ class Submissions < Core
 
         title("%s by %s in %s" % [submission.problem.name, submission.user.username, submission.problem.language])
 
+        comment_history = if current_user.guest? || !$flipper[:comment_history].enabled?(current_user)
```

Comment:
If I follow you, you mean something like…

```ruby
module Flipper
  class CommentHistory
    def comment_history(user:)
      if $flipper[:comment_history].enabled?(current_user)
        current_user.comments.recent(25)
      else
        []
      end
    end
  end
end
```

I'm not convinced for this particular scenario yet, but I want to make sure I understand what you're suggesting too. There are several non-user-visible changes that aren't turned off by the feature flag (e.g. schema changes, plumbing locals through erb views, coffeescript classes that are defined but unused) that it seems like, at least with how I implemented most of these features, that I don't feel like it would be easier or less complex to remove. Easier to test… I could be convinced that's true.

I'd love to hear about some examples where this approach has helped you. I assume it has, or I bet you wouldn't be suggesting it. 😄